frontend/src/metabase/entities/fields.js

```diff
@@ -191,6 +191,14 @@ const Fields = createEntity({
         updateIn(state, [fieldId, "remappings"], (existing = []) =>
           Array.from(new Map(existing.concat(remappings))),
         ),
+      // Can't import metabase/dashboard/FETCH_CARD_DATA because of circular dependency
+      ["metabase/dashboard/FETCH_CARD_DATA"]: (state, { payload }) => {
+        const fields = getIn(payload, ["result", "data", "cols"]);
+        return {
+          ...Fields.normalizeList(fields).entities[Fields.name],
```

Comment:
I really hate to have to normalize the data manually. I bet this isn't idiomatic, but I don't know how to get the data in the correct shape without normalizing it.

Under the hood, calling `normalizeList` will populate `uniqueId` which will be used by some of our selectors.
https://github.com/metabase/metabase/blob/00ce5700e31a98124b034e0586456e9ce261505c/frontend/src/metabase/selectors/metadata.js#L128

without that, `fields` won't be populated using the `getMetadata` selector.

Reply:
Let me take another look at it though